src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -415,16 +416,29 @@ private void initializeFromClientConfig(JedisClientConfig config) {
         fireAndForgetMsg.add(new CommandArguments(Command.CLIENT).add(Keyword.SETNAME).add(clientName));
       }
 
-      String libName = JedisMetaInfo.getArtifactId();
-      if (libName != null && validateClientInfo(libName)) {
-        fireAndForgetMsg.add(new CommandArguments(Command.CLIENT).add(Keyword.SETINFO)
-            .add(ClientAttributeOption.LIB_NAME.getRaw()).add(libName));
-      }
+      ClientSetInfoConfig setInfoConfig = config.getClientSetInfoConfig();
+      if (setInfoConfig == null) setInfoConfig = new ClientSetInfoConfig() { };
+
+      if (!setInfoConfig.disable()) {
+        String libName = JedisMetaInfo.getArtifactId();
```

Comment:
@vladvildanov I disagree. <strike>Firstly because I don't see further use. Even if we do, moving the logic won't be hard.

More importantly,</strike> ClientSetInfoConfig is an interface. So users can implement anything for `libName()`. This contradicts the _requirement_ allowing users only to add suffix.

Reply:
@sazzad16 In this case I see no purpose in public interface at all, if `ClientSetInfoConfig` would be an object it could accept suffix as external dependency, same with version and etc. 

IoC works perfectly here, object receives the data from outside, returns processed response and hides the logic behind libName and libVersion.

Reply:
Regarding this function, I don't even want to provide the ability to disable and set suffix (I haven't seen such a requirement yet), he should accurately pass the name and version of the client to facilitate server operation and maintenance and notification. Once we provide a modification interface, users may modify it incorrectly.

Reply:
> I haven't seen such a requirement yet

@yangbodong22011 There is a requirement. Sadly there is no publicly shareable doc available.

Reply:
@sazzad16 ok, let's move on.

Reply:
@vladvildanov Except the interface part, all the things you've said/suggested are true for this PR. It receives object with data (suffix, etc.) from outside and hides the logic (about libName and libVersion) behind. It is your suggestion in earlier comment that exposes the logic and makes it easy for the users to change the logic which goes against the requirement. Not sure how can I explain further. May be there is a language (java vs php) barrier here?

Reply:
@sazzad16 @yangbodong22011 format should be the following:
`jedis(Spring_Data_Redis_v3.1.1;JVM_v19.0.1;)`

As you can see from the format, **libNameSuffix shouldn't contain `(` and `)` characters** to allow easy parsing.